clang/lib/AST/Interp/Interp.cpp

```diff
@@ -332,6 +332,22 @@ bool CheckInitialized(InterpState &S, CodePtr OpPC, const Pointer &Ptr,
   return false;
 }
 
+bool CheckGlobalInitialized(InterpState &S, CodePtr OpPC, const Pointer &Ptr) {
+  if (Ptr.isInitialized())
+    return true;
+
+  const VarDecl *VD = cast<VarDecl>(Ptr.getDeclDesc()->asValueDecl());
+  if ((S.getLangOpts().CPlusPlus && !VD->hasConstantInitialization() &&
```

Comment:
Shouldn't this also be used in C? https://godbolt.org/z/G6Yv1T9Wd

Reply:
Yes, but in C we don't create global variables for `x`, so we don't try to load one either and don't reach this code.

Reply:
Should we have an assert that we're not in C mode and then drop the check for C++?